### PR TITLE
 JobSite-1104: corrected errors made in signup & login API. created job posting API

### PR DESCRIPTION
**Pull request by 04vignesh45**
corrected errors made in signup & login API. created job posting API

### CODE REVIEW

**Review by mdufreshers2023 — COMMENTED** [review]
Pointers to remember:
* First need to create a User - user have an access to comment on the job post and he able give like and dislike to the job post.
* Then admin and user will see all the job posts.
* Admin only can see their own post.
* User will have access to create, delete, update and Get the comments.
* Then compete the database part

File: handler/handler.go [hunk 1]
```diff
@@ -13,54 +14,200 @@ import (
...
 	}
 
-	// Generate JWT token
+	// Generate JWT token while signing up
 	token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
-		"id":    user.ID,
-		"email": user.Email,
+		"email":    user.Email,
+		"username": user.Username,
 	})
```
Comment (brocklesnarvj) [inline handler/handler.go:47]: Token-should be created on Login

File: handler/handler.go [hunk 2]
```diff
@@ -13,54 +14,200 @@ import (
...
 
 	tokenString, err := token.SignedString(models.SigningKey)
 	if err != nil {
-		return c.JSON(http.StatusInternalServerError, map[string]string{"message": "Failed to generate token"})
+		return c.JSON(http.StatusBadRequest, map[string]string{"warning": "Failed To Generate Token"})
 	}
```
Comment (brocklesnarvj) [inline handler/handler.go:52]: hi

File: handler/handler.go (hunk 1 above)
Comment (brocklesnarvj) [inline handler/handler.go:47]: Token-Creation should be implement in login handler

File: handler/handler.go [hunk 3]
```diff
@@ -13,54 +14,200 @@ import (
...
+	if !emailRegex.MatchString(updatedJob.Email) {
+		return c.JSON(http.StatusBadRequest, map[string]string{"warning": "Invalid Email Format."})
+	}
+
+	index := -1
+	for i, job := range models.Job {
+		if job.CompanyID == companyID {
+
+			index = i
+
```
Comment (brocklesnarvj) [inline handler/handler.go:173]: avoid unwanted newlines

File: router/router.go [hunk 4]
```diff
@@ -2,22 +2,29 @@ package router
...
-
-	// Router
-	e.POST("/signup", handler.Signup)
-	e.POST("/login", handler.Login)
-	e.POST("/posting",handler.Jobposting)
-	e.GET("/posting",handler.GetJobPostingDetails)
-	e.GET("posting/:id",handler.GetJobPostingByID)
-	e.PUT("posting/:id",handler.UpdateJob)
-	e.DELETE("posting/:id",handler.DeleteJob)
+    // Router
```
Comment (mdufreshers2023) [inline router/router.go:15]: Arrange the API's in order which is User and Admin

File: handler/handler.go [hunk 5]
```diff
@@ -2,207 +2,399 @@ package handler
...
-		if u.CompanyName == post.CompanyName {
-			return c.JSON(http.StatusBadRequest, map[string]string{"warning": "Company Name Already Exist"})
-		}
-		if u.Website == post.Website {
-			return c.JSON(http.StatusBadRequest, map[string]string{"warning": "Website Already Registered"})
-		}
-		if u.Email == post.Email {
-			return c.JSON(http.StatusBadRequest, map[string]string{"warning": "Email Already Exists"})
-		}
+	if post.CompanyName == "" || post.Website == "" || post.JobTitle == "" || post.JobType == "" || post.City == "" || post.State == "" || post.Country == "" || post.Email == "" || post.Description == "" {
```
Comment (mdufreshers2023) [inline handler/handler.go:140]: Need to separate the error(Which field is missing)

File: handler/handler.go [hunk 6]
```diff
@@ -2,207 +2,399 @@ package handler
...
 	}
-	if user.Username == ""{
-		return c.JSON(http.StatusBadRequest, map[string]string{"warning": "Username field should not be empty"})
+	if user.Username == "" {
+		return c.JSON(http.StatusBadRequest, map[string]interface{}{
+			"warning": "Username field should not be empty",
+			"status":  400,
+		})
 	}
 	if len(user.Password) < 8 {
```
Comment (mdufreshers2023) [inline handler/handler.go:42]: Need to hash the password

**Review by mdufreshers2023 — COMMENTED** [review]
Add the .env file for the database configuration.
Check all the status codes and correct the mismatched codes.
Need to hash the password
Move all the DB operations on repository

**Review by mdufreshers2023 — APPROVED** [review]

**Pull request merged**